app/services/umrdr/date_range_service.rb

```diff
@@ -0,0 +1,73 @@
+module Umrdr
+  class DateRangeService  
+
+    attr_reader :params
+    
+    def initialize(params)
+      @params = params
+    end  
+
+    def transform
+
+      if (params[:date_coverage_1_year].length>0)
+        yyyy1= params[:date_coverage_1_year].to_i 
+      else
+        yyyy2 = 0 #error 
+      end  
+      if (!params[:date_coverage_1_month].eql?("--"))
+        mm1 =  params[:date_coverage_1_month].to_i   
+      else
+        mm1 = 1    
```

Comment:
This is how Date method works, it returns 1st of the month and date if only year is entered